std/http/_io.ts

```diff
@@ -331,7 +332,11 @@ export async function readRequest(
   const req = new ServerRequest();
   req.conn = conn;
   req.r = bufr;
-  [req.method, req.url, req.proto] = firstLine.split(" ", 3);
+  [req.method, req.url, req.proto] = firstLine.split(" ", 3) as [
+    Method,
```

Comment:
It should support custom methods, right?

Reply:
If that's helpful for you, I think that's fine... Tho our server code should be changed to use those constants.

_Originally posted by @ry in https://github.com/denoland/deno/issues/5790#issuecomment-633070053_

Upon further consideration I'm not sure we should make the server or the request reader use the Method enum and instead opt for what GoLang's HTTP package has done (use string). While specifying an enumerated list of HTTP methods Go's HTTP package does not use that list to produce and parse Request structs as you can see here https://golang.org/src/net/http/request.go?s=3270:11920#L98

Reply:
Updated PR to reflect this GoLang.

Reply:
I would suggest to change the type of `method` field in `ServerRequest` to be like this:
```ts
type HttpMethod = HttpMethods | string;

export class ServerRequest {
  url!: string;
  method!:HttpMethod;
}
```

As you can see the type is not limited to the enum values but it can provide better intellisense when creating this class.

In general, I think we should not limit the bar of our expectations of type system as low as go's type system.

Reply:
@alisabzevari No, the intellisense will just treat it as `string` because that's what it is... there's no more information to be had. You want certain strings to have special high level significance which the actual code base doesn't care about -- you can't use the type system for this.

Custom methods are allowed and we don't handle them any differently, why do you want the intellisense to suggest we do?

Reply:
Yes, you are right. Sorry for the confusion.